src/config.ts

```diff
@@ -28,35 +32,44 @@ export function getConfig (): Config {
   if (prometheusMetrics && process.env.TRACING === 'false') {
     throw new TracingRequired('Prometheus')
   }
+  if (whitelistPath && allowIntrospection) {
+    throw new IntrospectionNotPermitted('whitelist')
+  }
```

Comment:
So we can't have whitelisting and allowIntrospection at the same time is forbidden. Introspection was allowed by default before. So what about simply having `allowIntrospection = !whitelistPath` to keep the previous behavior roughly the same and, not bother with this error scenario at all ?

Reply:
I feel the previous default was wrong, and introspection should be opt-in

Reply:
ApolloServer defaults to `true` when `process.env.NODE_ENV` is not `production`, but I just find these environmental variances cause more confusion than they solve.